### ISSUE TITLE
Make `validateLink`, `normalizeLink`, and `normalizeLinkText` pluggable

### ISSUE THREAD

**Issue by vberlier** [executablebooks/markdown-it-py#128]
I'm trying to parse markdown with data urls that aren't compatible with the built-in link validator. I would like to provide my own validator like the docstring of the [`validateLink` function](https://github.com/executablebooks/markdown-it-py/blob/11eb374b0b13a915693a9eeb1ef72dd496367ec7/markdown_it/common/normalize_url.py#L169) suggests but I don't see any documented way of doing that appart from monkey-patching the module.

```markdown
[my link](data:application/octet-stream;base64,H4sIAAAAAAAAAI2SwWrDMBBER3Kc2sqhPfXQT+m559JTrmFjr10RRTLSQqFfHxsi0gZaa0EHwdPMrBgDtNgk+80VgF+nRcNerFhOWKbF9uhCd0pmvjy2qKaQMpxHVaiTkPD1xU9GZaiA0f8wusALV621PKogjyrIs+aFzK146YI8uiBPidf9H6o/dl9j7r0W5mEixyK8VEU32LzTmfF8tp67SIO8JgmeD8do5zrBwHzEMHFcqtZg9zXrRBfGkXvUA7nEDbYDddaPqH2I8oks+XSTdNT3HOdUuzcS2nNMNnjAvOACT0G+1eQCAAA=)
```

I'm building a custom renderer so I'm using the `.parse()` method directly.

```python
p_open, inline, p_close = MarkdownIt().parse(...)
inline.children[0].type  # expected "link" but returns "text" instead
```

**Comment by chrisjsewell** [comment]
ah good point, I made `validateLink`, `normalizeLink`, and `normalizeLinkText` hardcoded, but I see your use case for allowing them to be "pluggable"

**Comment by vberlier** [comment]
That'd be great! I'm extracting files bundled in markdown. I don't really need the usual validation and normalization because I'm not producing any kind of html document. Being able to provide a custom `validateLink` and `normalizeLink` would help a lot!

**Comment by chrisjsewell** [comment]
@vberlier this should now be fixed by #135 but obviously let me know if there is still any issues

**Comment by vberlier** [comment]
Thanks! This should be perfect. Do you plan on releasing this soon?
